att_err_mon.py

```diff
@@ -1,158 +1,150 @@
+                        default=".")
+    parser.add_argument("--recent-start")
     opt = parser.parse_args()
     return opt
 
 
-def get_obs_table(start, stop, msf):
+def get_obs_table(start, stop):
     """
-    Make a data table of obsids with MS filter status, one shot magnitudes, and att errors
-    Only include observations that match the supplied MS filter status
+    Make a data table of obsids with one shot magnitudes and att errors
 
     :param start: start time for dwell range to fetch
     :param stop: stop time for range
-    :param msf: multiple star filter status ('ENAB' or 'DISA')
     :returns: astropy table of observation data
     """
-    m_filter_telem = maude.get_msids('AOACIMSS', start, stop)
-    aoacimss = m_filter_telem['data'][0]
-    dwells = events.dwells.filter(start=start, stop=stop)
+    manvrs = events.manvrs.filter(start=start, stop=stop, n_dwell__gte=1)
     obs_data = []
-    for d in dwells:
+    for m in manvrs:
         obs = {}
         try:
-            obsid = d.get_obsid()
+            obsid = m.get_obsid()
         except ValueError:
             obsid = 0
         if obsid is None:
             obsid = 0
-        #print obsid, d.start
-        # for data earlier than the existence of the AOACIMSS msid,
-        # the value can be assumed to be 'ENAB'
-        if d.manvr.kalman_start < '2015:252':
-            obs['gui_ms'] = 'ENAB'
-        elif len(aoacimss['times']) and (d.tstop > aoacimss['times'][-1]):
-            continue
-        else:
-            mid_kalman = ((DateTime(d.manvr.kalman_start).secs
-                            + DateTime(d.stop).secs) / 2)
-            kal_idx = (np.searchsorted(aoacimss['times'],
-                                        mid_kalman))
-            obs['gui_ms'] = aoacimss['values'][kal_idx]
-
-        # if the dwell isn't for the type of data we're looking for, just continue
-        if obs['gui_ms'] != msf:
-            continue
 
-        # Get one shot and maneuver to for *next* obsid
-        n = d.get_next()
-        if not n:
-            continue
-        try:
-            obs['next_obsid'] = n.get_obsid()
-        except ValueError:
-            obs['next_obsid'] = 0
+        # If there is no next maneuver, we're done as there's no interval percentiles
+        if m.get_next() is False:
+            break
 
-        obs['date'] = d.manvr.kalman_start
-        obs['datestop'] = d.stop
-        obs['time'] = DateTime(d.manvr.kalman_start).secs
-        obs['timestop'] = d.tstop
-
-        obs['manvr_angle'] = n.manvr.angle
-        obs['one_shot'] = n.manvr.one_shot
         obs['obsid'] = obsid
+        obs['date'] = m.start
+        obs['time'] = DateTime(m.start).secs
+        obs['manvr_angle'] = m.angle
+        obs['one_shot'] = m.one_shot
+        obs['one_shot_yaw'] = m.one_shot_yaw
+        obs['one_shot_pitch'] = m.one_shot_pitch
+        obs['dwell_duration'] = DateTime(m.get_next().start).secs - DateTime(m.npnt_start).secs
```

Comment:
Is there a reason you can't use `m.next_nman_start`?

Reply:
I don't think so.  There are a lot of attributes in there; is that one better?

Reply:
Yes, `next_nman_start` is better since it doesn't require that there be a next entry in the database and it is more explicit. The `start` attribute refers to the actual start of the maneuver. Being fussy, what we want here is the start of NMAN for the next manuever.

The maneuver events are only put into the database if they are "complete", meaning that the "next" maneuver has been seen in telemetry and the `next_man_start` field can be filled in correctly.

Reply:
OK. Though I just had this loop breaking a few lines up if get_next() was null so they somewhat became equivalent?  Also, regarding NMAN transition, the times used for the range for the point/roll errors are more explicit anyway (and trim); this rough dwell duration is really just used for the 1000s filter (though if we want something else would be good to know anyway).

Reply:
You shouldn't need the `get_next()` test above if you don't use `get_next()` here.  More win for code simplicity.